api/v1alpha1/amaltheasession_children.go

```diff
@@ -0,0 +1,220 @@
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// StatefulSet returns a AmaltheaSession StatefulSet object
+func (cr *AmaltheaSession) StatefulSet() appsv1.StatefulSet {
+	labels := labelsForAmaltheaSession(cr.Name)
+	replicas := int32(1)
+	if cr.Spec.Hibernated {
+		replicas = 0
+	}
+
+	session := cr.Spec.Session
+	pvc := cr.PVC()
+
+	sessionContainer := v1.Container{
+		Image:           session.Image,
+		Name:            "session",
+		ImagePullPolicy: v1.PullIfNotPresent,
+
+		Ports: []v1.ContainerPort{{
+			ContainerPort: session.Port,
+			Name:          "session-port",
```

Comment:
Same question as above, should we use a constant and maybe it's content be dependent on the sessionContainer name.